packages/retail-ui-extensions/src/extension-api/scanner-api/scanner-api.ts

```diff
@@ -0,0 +1,24 @@
+import type {RemoteSubscribable} from '@remote-ui/async-subscription';
+
+/** The scanner source the POS device supports */
+export type SupportedSource = 'camera' | 'external' | 'embedded';
+
+export interface ScannerSubscriptionResult {
+  /** The string data from the last scanner event received */
+  data?: string;
```

Comment:
What does a case look like where `data` would be undefined? Wouldn't a successful scan always provide something?

Reply:
Agreed, this shouldn't be undefined.